node/pod.go

```diff
@@ -68,9 +69,12 @@ func (pc *PodController) createOrUpdatePod(ctx context.Context, pod *corev1.Pod)
 
 	// We do this so we don't mutate the pod from the informer cache
 	pod = pod.DeepCopy()
-	if err := populateEnvironmentVariables(ctx, pod, pc.resourceManager, pc.recorder); err != nil {
-		span.SetStatus(err)
-		return err
+	if pc.canResolveEnvVars() {
+		err := podutils.PopulateEnvironmentVariables(ctx, pod, pc.recorder, pc.configMapLister, pc.secretLister, pc.serviceLister)
```

Comment:
Maybe this is dumb question, but how do we know that these are consistent / up to date?